lib/web/saml.go

```diff
@@ -96,7 +96,7 @@ func (h *Handler) samlACS(w http.ResponseWriter, r *http.Request, p httprouter.P
 		return client.LoginFailedRedirectURL
 	}
 
-	response, err := h.cfg.ProxyClient.ValidateSAMLResponse(r.Context(), samlResponse)
+	response, err := h.cfg.ProxyClient.ValidateSAMLResponse(r.Context(), samlResponse, p.ByName("connector"))
```

Comment:
I just improved the error message in an inner validation function a bit. However, we can't know if an empty string is invalid here. To not break backwards compatibility with SSO configurations which don't provide this parameter, we allow the default value of an empty string to be passed, however, we only allow IdP-initiated logins if this parameter is passed, otherwise they will fail.